File: src/roles.js
export const ROLE_INITIATOR = "initiator";
/** @deprecated JWT/DB legacy alias — treated as initiator */
export const ROLE_DONOR = "donor";
export const ROLE_COORDINATOR = "coordinator";

export function isInitiatorRole(role) {
  return role === ROLE_INITIATOR || role === ROLE_DONOR;
}

export function normalizeRole(role) {
  if (role === ROLE_COORDINATOR) {
    return ROLE_COORDINATOR;
  }
  if (isInitiatorRole(role)) {
    return ROLE_INITIATOR;
  }
  return ROLE_INITIATOR;
}

File: src/main/java/org/sharingbridge/integration/auth/Roles.java
package org.sharingbridge.integration.auth;

public final class Roles {

    public static final String ROLE_INITIATOR = "initiator";
    /** JWT/DB legacy alias — treated as initiator. */
    @Deprecated
    public static final String ROLE_DONOR = "donor";
    public static final String ROLE_COORDINATOR = "coordinator";

    private Roles() {}

    public static boolean isInitiatorRole(String role) {
        return ROLE_INITIATOR.equals(role) || ROLE_DONOR.equals(role);
    }

    public static String normalizeRole(String role) {
        if (ROLE_COORDINATOR.equals(role)) {
            return ROLE_COORDINATOR;
        }
        if (isInitiatorRole(role)) {
            return ROLE_INITIATOR;
        }
        return ROLE_INITIATOR;
    }
}
